test/custom_log_gateway_test.go

```diff
@@ -0,0 +1,190 @@
+	SetUpTest(t)
+
+	SpinUpTheModule()
+	defer CleaningUpAtTheEnd()
+	WriteAMessageToTheApiAndExpect(200, withCorrectApiKey())
+}
+
+func TestLogCannotBeSubmittedToApiWithoutApiKey(t *testing.T) {
+	SetUpTest(t)
+
+	SpinUpTheModule()
+	defer CleaningUpAtTheEnd()
+	WriteAMessageToTheApiAndExpect(403, "")
+
+
+}
+
+
+func TestThatQueueHasServerSideEncryptionEnabled(t *testing.T) {
+	SetUpTest(t)
+
+	SpinUpTheModule()
+	defer CleaningUpAtTheEnd()
+	VerifyThatQueueEncryptionIsEnabled(t)
+}
+
+func SetUpTest(t *testing.T) {
+	_t = t
+}
+
+func CleaningUpAtTheEnd() string {
+	return terraform.Destroy(_t, _terraformOptions)
+}
+
+func withCorrectApiKey() string {
+	apiKey := terraform.Output(_t, _terraformOptions, "custom_logging_api_key")
+	return apiKey
+}
+
+func VerifyThatMessageWasPlacedOnQueue() {
+	sess, _ := session.NewSession(&aws.Config{Region: aws.String("eu-west-2")})
+
+	sqsService := sqs.New(sess)
+
+	queueUrl := terraform.Output(_t, _terraformOptions, "custom_log_queue_url")
+
+	result, err := sqsService.ReceiveMessage(&sqs.ReceiveMessageInput{
+		AttributeNames: []*string{
+			aws.String(sqs.MessageSystemAttributeNameSentTimestamp),
+		},
+		MessageAttributeNames: []*string{
+			aws.String(sqs.QueueAttributeNameAll),
+		},
+		QueueUrl:            &queueUrl,
+		MaxNumberOfMessages: aws.Int64(1),
+		VisibilityTimeout:   aws.Int64(20),  // 20 seconds
+		WaitTimeSeconds:     aws.Int64(0),
+	})
+
+	if err != nil {
+		fmt.Println("***Error:***", err.Error())
+		return
+	}
+
+	if len(result.Messages) == 0 {
+		_t.Fatalf("***Received no messages***")
+		_t.Fail()
+		return
+	}
+
+	expectedMessageBodyBytes, _ := json.Marshal(map[string]string{
+		"foo": "bar",
+	})
+
+	expectedMessageBody := string(expectedMessageBodyBytes)
+
+
+	messageBody := reformatJsonString(*result.Messages[0].Body)
+
+
+	if messageBody != expectedMessageBody {
+		log.Println("***expected message***:")
+		log.Println(expectedMessageBody)
+		log.Println("***but got***:")
+		log.Println(messageBody)
+		_t.Fail()
+	}
+}
+
+func reformatJsonString(theThing string) string {
+	var messageBodyMap map[string]interface{}
+	err := json.Unmarshal([]byte(theThing), &messageBodyMap)
+
+	if err != nil {
+		log.Println("***Unable to reformat json***")
+		log.Println(err.Error())
+		log.Println(theThing)
+	}
+
+	messageBody, _ := json.Marshal(messageBodyMap)
+
+	return string(messageBody)
+}
+
+func WriteAMessageToTheApiAndExpect(code int, apiKey string) {
+	loggingEndpointPath := terraform.Output(_t, _terraformOptions, "logging_endpoint_path")
+
+	requestBody, _ := json.Marshal(map[string]string{
+		"foo": "bar",
+	})
+
+	_, err := http_helper.HTTPDoWithRetryE(_t,
+		"POST",
+		loggingEndpointPath,
+		requestBody,
+		map[string]string{"Content-Type": "application/json", "X-API-KEY": apiKey},
+		code,
+		5,
+		retryDelay,
+		nil,
+	)
+
+	if err != nil {
+		_t.Fatalf("***Api did not return code '%d'***", code)
+		_t.Fail()
+	}
+}
+
+func SpinUpTheModule(){
```

Comment:
I like the descriptive methods, would it be possible to refactor the other test to use these?